### ISSUE TITLE
help: Error during Cloudflare Pages deployment after installing @nuxt/og-image module

### ISSUE THREAD

**Issue by erados** [nuxt-modules/og-image#263]
### 📚 What are you trying to do?

**Description:**

I'm deploying a Nuxt 3 application to Cloudflare Pages using the following command:

```bash
NITRO_PRESET=cloudflare-pages nuxt build && wrangler pages deploy dist/
```

The deployment was working as expected until I installed the `@nuxt/og-image` module. After installing the module, the following error occurs during the build:

```
✘ [ERROR] Build failed with 1 error:

_worker.js/chunks/_/index_bg.mjs:1:15: ERROR: No loader is configured for ".wasm" files:
_worker.js/wasm/index_bg-dd4dd8881e2df4e6.wasm
```

It seems that the build process is failing due to an issue related to loading `.wasm` files.

---

**Steps to reproduce:**

1. Add `@nuxt/og-image` module to the Nuxt 3 project.
2. Deploy to Cloudflare Pages using the following command:

   ```bash
   NITRO_PRESET=cloudflare-pages nuxt build && wrangler pages deploy dist/
   ```

3. Observe the build failure with the `.wasm` file loader error.

---

**Expected behavior:**

The build should succeed, and the project should be deployed to Cloudflare Pages without errors, even with the @nuxt/og-image module installed.

---

**Actual behavior:**

The build fails with an error related to the `.wasm` file:

```
_worker.js/chunks/_/index_bg.mjs:1:15: ERROR: No loader is configured for ".wasm" files:
_worker.js/wasm/index_bg-dd4dd8881e2df4e6.wasm
```

---

**Environment:**

- node v20.17.0
  "dependencies": {
    "@nuxt/ui": "^2.18.4",
    "@nuxtjs/i18n": "^8.5.1",
    "@nuxtjs/supabase": "^1.4.0",
    "firebase": "^10.13.1",
    "nuxt": "^3.13.0",
    "nuxt-og-image": "^3.0.2",
    "vue": "latest",
    "vue-i18n": "9",
    "wrangler": "^3.78.7"
  },
  "packageManager": "yarn@1.22.22+sha512.a6b2f7906b721bba3d67d4aff083df04dad64c399707841b7acf00f6b133b7ac24255f2652fa22ae3534329dc6180534e98d17432037ff6fd140556e2bb3137e",
  "devDependencies": {
    "@nuxtjs/tailwindcss": "^6.12.1",
    "sass-embedded": "^1.77.8",
    "supabase": "^1.191.3",
    "typescript": "^5.5.4",
    "vue-tsc": "^2.1.6"
  }

---

**Possible cause:**
It seems that `.wasm` files generated during the build process are not being handled properly by the Cloudflare Pages build environment.

Could someone provide guidance on how to configure the build process to handle `.wasm` files correctly when deploying to Cloudflare Pages?

### 🔍 What have you tried?
I've set the compatibilities to 'node' so that there is no need for 'wasm'.

```json
{
  "ogImage": {
    "compatibility": {
      "prerender": {
        "satori": "node",
        "resvg": "node",
        "css-inline": "node",
        "sharp": "node",
        "chromium": false
      }
    }
  }
}
```

**Comment by erados** [comment]
I modified prerender to runtime in the Nuxt config, and it resolved the issue. 

However, since I’m using dynamic routing with user-generated content IDs ([id].vue), I can’t prerender all the og-images for the content, as each new content would require a build.

**Comment by erados** [comment]
The current bundle size is as follows:

```
Before: Total upload: 1988.13 KiB / gzip: 471.47 KiB  
After: Total upload: 4673.76 KiB / gzip: 1325.43 KiB
```

Build and deploy using:

```
NITRO_PRESET=cloudflare-pages yarn build && wrangler deploy ./dist/_worker.js --dry-run --name a
```

**Comment by harlan-zw** [comment]
Hi, thanks for the issue. I'll address just the core issue you have raised which is the .wasm entry is not being transpiled.

I've pushed up a fix in 3.0.3 that may resolve it.  It's a strange issue though as I wasn't able to replicate using the playground.

Let me know if you still have issues.

**Comment by erados** [comment]
The issue has been resolved. Thanks 👍
